extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -19,14 +20,45 @@ export abstract class ToolBase implements ITool {
 	abstract type: ToolType;
 	abstract homePage: string;
 	abstract autoInstallSupported: boolean;
-	public abstract install(): Promise<void>;
+	abstract installationCommands: Command[];
 	protected abstract getVersionFromOutput(output: string): SemVer | undefined;
-	protected abstract readonly versionCommand: string;
+	protected abstract readonly versionCommand: Command;
+
+	protected get installationPath(): Promise<string | null> {
+		return new Promise<string | null>((resolve, _reject) => {
```

Comment:
If you're returning a Promise, mark the method as async

Reply:
actually - null usage is correct beasuse in my understanding undefined should be used for cases where a parameter or optional value is not passed, but as you said we have not been consistent with this in our code base so changing to undefined.